src/essential/ad.jl

```diff
@@ -77,13 +77,17 @@ Find the autodifferentiation backend of the algorithm `alg`.
 """
 getADbackend(spl::Sampler) = getADbackend(spl.alg)
 getADbackend(::SampleFromPrior) = ADBackend()()
```

Comment:
Are these still needed? Or could we remove them and specialize `getADbackend(::SamplingContext{<:Sampler})` and `getADbackend(::SamplingContext{SampleFromPrior})`? I think mentally it's easier if `getADbackend` only operates on contexts.

Reply:
Yeah I can try doing that! I was worried this would break too many tests and discourage me from completing the PR, but now that the tests are passing I can see if it's worth it.